src/ProjectManagement/Utility/BuildIntegratedProjectUtility.cs

```diff
@@ -79,5 +77,77 @@ public static ExternalProjectReference ConvertProjectReference(BuildIntegratedPr
         {
             return new ExternalProjectReference(reference.Name, reference.PackageSpecPath, reference.ExternalProjectReferences);
         }
+
+        public static IReadOnlyList<PackageIdentity> GetOrderedProjectDependencies(BuildIntegratedNuGetProject buildIntegratedProject)
+        {
+            var results = new List<PackageIdentity>();
+
+            var lockFilePath = BuildIntegratedProjectUtility.GetLockFilePath(buildIntegratedProject.JsonConfigPath);
+            var lockFileFormat = new LockFileFormat();
+
+            // Read the lock file to find the full closure of dependencies
+            if (File.Exists(lockFilePath))
+            {
+                var lockFile = lockFileFormat.Read(lockFilePath);
+
+                var dependencies = new HashSet<PackageDependencyInfo>(PackageIdentity.Comparer);
+
+                foreach (var target in lockFile.Targets)
+                {
+                    foreach (var targetLibrary in target.Libraries)
+                    {
+                        var identity = new PackageIdentity(targetLibrary.Name, targetLibrary.Version);
+                        var dependency = new PackageDependencyInfo(identity, targetLibrary.Dependencies);
+                        dependencies.Add(dependency);
+                    }
+                }
+
+                // Sort dependencies
+                var sortedDependencies = SortPackagesByDependencyOrder(dependencies);
+                results.AddRange(sortedDependencies);
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Order dependencies by children first.
+        /// </summary>
+        private static IReadOnlyList<PackageDependencyInfo> SortPackagesByDependencyOrder(IEnumerable<PackageDependencyInfo> packages)
+        {
+            var sorted = new List<PackageDependencyInfo>();
+            var toSort = packages.Distinct().ToList();
+
+            while (toSort.Count > 0)
+            {
+                // Order packages by parent count, take the child with the lowest number of parents and remove it from the list
+                var nextPackage = toSort.OrderBy(package => GetParentCount(toSort, package.Id))
```

Comment:
Could this be re-written to Order by `GetChildCount` i.e. count the number of dependencies that are present in the list and reverse that? Re-ordering every time seems pretty expensive.